### PR TITLE
New backend status AwaitingCloudQuota [BW-1012]

### PR DESCRIPTION
**Pull request by aednichols**
Since this is fairly backend-specific I've implemented as a `cromwell.backend.google.pipelines.common.api.RunStatus` instead of a `cromwell.core.ExecutionStatus`. It also reduces the scope of changes. Feedback welcome.

Looks like this in metadata requests:
```
"calls": {
    "sleepy_sleep.sleep": [
      {
        "preemptible": false,
        "executionStatus": "Running",
        "stdout": "gs://cloud-cromwell-dev-self-cleaning/cromwell_execution/ci/sleepy_sleep/058bff35-4a55-4c0f-9113-0885f4119cd9/call-sleep/stdout",
        "backendStatus": "AwaitingCloudQuota",
        "compressedDockerSize": 28566425,
        "commandLine": "sleep 180;\nls -la",
        "shardIndex": -1,
        "jes": {
          "executionBucket": "gs://cloud-cromwell-dev-self-cleaning/cromwell_execution/ci",
          "endpointUrl": "https://lifesciences.googleapis.com/",
          "googleProject": "broad-dsde-cromwell-dev"
        },
```
The Terra UI workflow dashboard currently makes calls that look like
```
https://rawls.dsde-dev.broadinstitute.org/api/workspaces/general-dev-billing-account/anichols-post-ppw/submissions/a7cfb487-9c30-4fd7-b10f-19d6f3c1d192/workflows/f52e4e4b-0299-44b7-a16f-904d2ee3f1e9
  ?includeKey=end
  &includeKey=executionStatus
  &includeKey=failures
  &includeKey=start
  &includeKey=status
  &includeKey=submittedFiles:workflow
  &includeKey=workflowLog
  &includeKey=workflowName
  &includeKey=callCaching:result
  &includeKey=callCaching:effectiveCallCachingMode
```
so it would be easy to add a `&includeKey=backendStatus` and logic to evaluate something like
```
if (executionStatus == "Running" && backendStatus == "AwaitingCloudQuota") {
  msg = "Trying to run but needs quota"
}
```

### CODE REVIEW

**Comment by geoffjentry** [comment]
download

File: supportedBackends/google/pipelines/v2beta/src/main/scala/cromwell/backend/google/pipelines/v2beta/api/request/GetRequestHandler.scala [hunk 1]
```diff
@@ -164,4 +166,26 @@ trait GetRequestHandler { this: RequestHandler =>
...
+
+  // Future enhancement: parse as `com.google.api.services.lifesciences.v2beta.model.DelayedEvent` instead of
+  // generic `Event` and take advantage of `getMetrics` which has a string array of problem quota(s):
+  //   "metrics": [
+  //     "CPUS"
+  //   ]
+  private def isQuotaDelayed(events: List[Event]): Boolean = {
+    events.sortBy(_.getTimestamp).reverse.headOption match {
+      case Some(event) =>
+        quotaMessages.exists(event.getDescription.contains)
```
Comment (cahrens) [inline supportedBackends/google/pipelines/v2beta/src/main/scala/cromwell/backend/google/pipelines/v2beta/api/request/GetRequestHandler.scala:178]: This is shorthand for

`quotaMessages.exists(message => event.getDescription.contains(message))`?

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,31 @@
 # Cromwell Change Log
 
+## 75 Release Notes
+
+### New `AwaitingCloudQuota` backend status
+
+For Cloud Life Sciences v2beta only.
```
Comment (cahrens) [inline CHANGELOG.md:7]: Nice release notes!

**Review by cahrens — APPROVED** [review]

File: supportedBackends/google/pipelines/v2beta/src/main/scala/cromwell/backend/google/pipelines/v2beta/api/request/GetRequestHandler.scala (hunk 1 above)
Comment (aednichols) [reply supportedBackends/google/pipelines/v2beta/src/main/scala/cromwell/backend/google/pipelines/v2beta/api/request/GetRequestHandler.scala:178]: Yes, or the intermediate level of shortening
```
quotaMessages.exists(event.getDescription.contains(_))
```

**Comment by aednichols** [comment]
Tested end-to-end in dev and confirmed that Terra UI and Job Manager don't regress. I used the Firefox edit request feature to add an include key for `backendStatus` and it works as expected.

Screen Shot 2022-01-20 at 6 14 44 PM

**Review by salonishah11 — APPROVED** [review]

**Pull request merged**